posthog-core/src/index.ts

```diff
@@ -724,25 +724,54 @@ export abstract class PostHogCore extends PostHogCoreStateless {
   }
 
   protected setupBootstrap(options?: Partial<PostHogCoreOptions>): void {
-    if (options?.bootstrap?.distinctId) {
-      if (options?.bootstrap?.isIdentifiedId) {
-        this.setPersistedProperty(PostHogPersistedProperty.DistinctId, options.bootstrap.distinctId)
+    const bootstrap = options?.bootstrap
+    if (!bootstrap) {
+      return
+    }
+
+    // bootstrap options are only set if no persisted values are found
+    // this is to prevent overwriting existing values
```

Comment:
this code lives in the core package so I assume this affects node and web as well.
not sure if node and web will have storage (likely memory storage), but IMO the change still makes sense.
if you bootstrap a flag and either there's a value or a new value comes from the decide API, the cached value or the new value from flags have precedence over the bootstrap flag.

Reply:
hmmm, I assume that's checked with @PostHog/team-feature-success my assumption would have been that client provided bootstrap values would override remote values - like we do with config. E.g. disabled with client config overrules enabled with remote config.

but that's only asked with no context - I don't know the right answer

Reply:
https://github.com/PostHog/posthog-js-lite/pull/260/files that would work for RN only (would need fix tests) but if that's the expectation, happy to do it, I still think it's strange as per [docs](https://github.com/PostHog/posthog-js-lite/pull/259#issuecomment-2346441186), and as I have used bootstrap flags in other tools.
@Phanatic @neilkakkar can you decide on the behaviour here? thx